### PR TITLE
Avoid false positives when linting for pyspark patterns

### PR DESCRIPTION
**Pull request by ericvergnaud**
## Changes
Check originating module when linting for pyspark calls

### Linked issues
None

### Functionality
None

### Tests
- [x] ran unit tests

### CODE REVIEW

File: tests/unit/source_code/linters/test_pyspark.py [hunk 1]
```diff
@@ -277,9 +278,9 @@ def test_spark_sql_fix(migration_index):
                     code='direct-filesystem-access',
                     message='The use of direct filesystem references is deprecated: '
                     "s3a://your_bucket_name/your_directory/",
-                    start_line=0,
+                    start_line=1,
                     start_col=1,
-                    end_line=2,
+                    end_line=3,
                     end_col=59,
```
Comment (asnare) [inline tests/unit/source_code/linters/test_pyspark.py:284]: Not a comment on this PR per se, but more a general observation: should't these warnings refer to the problematic argument?

In this case the positioning information should really be something like: 3:19 thru 3:59

Humans can understand the current positioning information, even if more accurate highlighting is possible. But will this start to cause problems when it comes to automatic rewriting/fixing?

Comment (ericvergnaud) [reply tests/unit/source_code/linters/test_pyspark.py:284]: I agree with you, and it's been done with some linters, but let's have field engineers opine on this?

Comment (asnare) [reply tests/unit/source_code/linters/test_pyspark.py:284]: Yup; this was intended as the start of a conversation. 👍

**Review by nfx — APPROVED** [review]
nice

**Pull request merged**